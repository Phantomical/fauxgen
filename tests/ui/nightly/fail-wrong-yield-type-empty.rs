#![feature(generators)]

#[fauxgen::generator(yield = i32)]
fn gen() {
    yield;
}

fn main() {}
