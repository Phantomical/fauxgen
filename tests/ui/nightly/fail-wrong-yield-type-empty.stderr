error[E0308]: mismatched types
 --> tests/ui/nightly/fail-wrong-yield-type-empty.rs:5:5
  |
3 | #[fauxgen::generator(yield = i32)]
  |                      ----------- arguments to this method are incorrect
4 | fn gen() {
5 |     yield;
  |     ^^^^^ expected `i32`, found `()`
  |
help: the return type of this call is `()` due to the type of the argument passed
   --> tests/ui/nightly/fail-wrong-yield-type-empty.rs:3:1
    |
3   | #[fauxgen::generator(yield = i32)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
4   | fn gen() {
5   |     yield;
    |     ----- this argument influences the return type of `yield_`
note: method defined here
   --> src/detail/token.rs
    |
    |     pub async fn yield_(self: Pin<&Self>, value: Y) -> A {
    |                  ^^^^^^
    = note: this error originates in the macro `yield` (in Nightly builds, run with -Z macro-backtrace for more info)
